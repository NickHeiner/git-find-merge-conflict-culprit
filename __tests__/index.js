const execa = require('execa');
const tempy = require('tempy');
const cpy = require('cpy');
const path = require('path');
const packageJson = require('../package');
const globby = require('globby');
const fs = require('fs');
const _ = require('lodash');
const log = require('nth-log');

test('git-find-files-not-modified-on-feature-branch', async () => {
  const testDir = await tempy.directory();

  log.debug({testDir}, 'Created test directory');

  await cpy('', testDir, {
    cwd: path.join(__dirname, '__fixtures__'),
    parents: true
  });
  await execa('mv', ['git', '.git'], {cwd: testDir});

  const binPath = path.resolve(__dirname, '..', packageJson.bin['git-find-files-not-modified-on-feature-branch']);

  await execa(binPath, [
    '--compareBranch', 'compare',
    '--baseBranch', 'base',
    '--dry', 'false'
  ], {cwd: testDir});

  const filePaths = await globby(path.join('**', '*.*'), {cwd: testDir});
  const fileContents = _(filePaths)
    .map(filePath => [filePath, fs.readFileSync(path.join(testDir, filePath), 'utf8')])
    .fromPairs()
    .value();

  expect(fileContents).toMatchSnapshot();
});