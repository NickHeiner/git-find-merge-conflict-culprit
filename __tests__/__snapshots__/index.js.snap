// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`git-find-files-not-modified-on-feature-branch 1`] = `
Object {
  "added.txt": "new file
",
  "new-compare-file.txt": "new compare file
",
  "rename-on-base-but-not-compare.txt": "to rename but should stay with the original name on compare

Modified on compare",
  "renamed-on-base-but-not-compare.txt": "to rename but should stay with the original name on compare",
  "renamed.txt": "To rename",
  "to-leave-unchanged.txt": "To leave unchanged",
  "to-modify.txt": "changed
",
}
`;
